examples/bigcommerce/lib/bigcommerce/api.ts

```diff
@@ -130,7 +130,5 @@ export async function getProduct(id?: number): Promise<ProductFragment> {
 
   const [productEdge] = result.data.site.products.edges
 
-  if (productEdge == null) throw new Error(`Product with ID "${id}" not found.`)
-
-  return productEdge.node
+  return productEdge?.node
```

Comment:
The type of this function is now wrong. The real type is now `Promise<ProductFragment | undefined>`. I try to avoid `undefined`, though, and would suggest you make the type `Promise<ProductFragment | null>` and have the return type be `productEdge?.node ?? null`.

The reason TypeScript didn't catch this was because of the array access. Since TypeScript doesn't check for array bounds, array access is "unsafe"

Reply:
We should turn on `noUncheckedIndexedAccess`. It seems that it's a new option since TypeScript 4.1 that checks for this

Reply:
Check it out in action here: https://www.typescriptlang.org/play?noUncheckedIndexedAccess=true#code/MYewdgzgLgBGCuBbARgUwE4RgXhgbQEYAaGAJhIGYBdAKBtElmQEMATHOJNTPAFlvrgIIADaoAdCJABzABQt2AahgEAlDSA

Reply:
Yet more info: https://github.com/microsoft/TypeScript/pull/39560